File: hw3/shield.js
class Shield {
    constructor(style, material, color, weight, thrown) {
        this.style = 'style'
        this.material = 'material'
        this.color = 'color'
        this.weight = 0
        this.thrown = false
    }

    throwShield() {
        this.thrown = true
    }

    get style() {
        return this.style
    }

    get material() {
        return this.material
    }
    
    get color() {
        return this.color
    }

    get weight() {
        return this.weight
    }

    get thrown() {
        return this.thrown
    }
}

File: cppArchy/src/shield.cpp
#include "shield.h"
#include <stdexcept>

namespace shield {

    Shield::Shield() : Armor("steel", 0.0) //default shield
    {
        m_style = "kite";
        m_color = "blue";
        m_weight = 10;
        m_thrown = false;
    }

    Shield::Shield(std::string material, double defense, 
                std::string style, std::string color, int weight, bool thrown) 
        : Armor(material, defense) //base class constructor
    {
        m_style = style;
        m_color = color;
        m_weight = weight;
        m_thrown = thrown;
    }

    void Shield::throwShield() {
        m_thrown = true;
    }

    std::string Shield::style() {
        return m_style;
    }

    std::string Shield::color() {
        return m_color;
    }

    int Shield::weight() {
        return m_weight;
    }

    bool Shield::thrown() {
        return m_thrown;
    }
}